pkg/server/api/audit/audit.go

```diff
@@ -0,0 +1,72 @@
+package audit
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/spire-api-sdk/proto/spire/api/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type Log interface {
+	AddFields(logrus.Fields)
+	Emit(logrus.Fields)
+	EmitBatch(*types.Status, logrus.Fields)
+	EmitError(error)
+}
+
+type log struct {
+	fields logrus.Fields
+	log    logrus.FieldLogger
+}
+
+func New(l logrus.FieldLogger) Log {
+	return &log{
+		log: l.WithFields(logrus.Fields{
+			"type": "audit",
+			// It is success by default, erros must change it
+			"status": "success",
+		}),
+		fields: logrus.Fields{},
+	}
+}
+
+func (l *log) AddFields(fields logrus.Fields) {
+	for key, value := range fields {
+		l.fields[key] = value
+	}
+}
+
+func (l *log) Emit(fields logrus.Fields) {
+	l.log.WithFields(l.fields).WithFields(fields).Info("Audit log")
+}
+
+func (l *log) EmitError(err error) {
+	fields := fieldsFromError(err)
+	l.log.WithFields(l.fields).WithFields(fields).Info("Audit log")
```

Comment:
Should the message string that we emit with these log lines be defined as a constant?

I also wonder what is the best thing to put here ... We already have the type=audit field attached ... can or should the message be something more useful?

Reply:
I'm not sure what another thing to add as message, as you I think we already have `type` we can make each function to set a message, or we can set a 'SUCCESS' / 'FAILURE' based on if error is present